src/client.rs

```diff
@@ -288,24 +323,80 @@ impl ApiClient {
             .expect_json()
     }
 
-    pub fn start_analysis(&self, project_id: u32, files: Vec<&Path>) -> ApiResult<ApiAnalysisJobResponse> {
-        let form= files
+    pub fn get_branches_for_project(&self, project_id: u32) -> ApiResult<Vec<ApiBranch>> {
+        self.api_get(&["x", "projects", &project_id.to_string(), "branches"])
+            .expect_success()
+            .expect_json()
+    }
+
+    pub fn query_branches_for_project(&self, project_id: u32, branch_name: &str) -> ApiResult<Vec<ApiBranch>> {
+       self.get_branches_for_project(project_id).map(
+            |branches| branches.into_iter().filter(|branch| branch.name.contains(branch_name)).collect()
+        )
+    }
+
+    pub fn start_analysis(&self, project_context: ProjectContext, branch_name: Option<String>, files: Vec<&Path>) -> ApiResult<ApiAnalysisJobResponse> {
+        let branch_name_string = match branch_name {
+            Some(name) => name,
+            None => "".to_string()
+        };
```

Comment:
Could use `.unwrap_or_else(|| "".to_string())` 

`unwrap_or` and `unwrap_or_else` seem to be the equivalents of Scala Option's `.getOrElse` (one passes a value, one passes a closure)

Reply:
Actually `.unwrap_or_default()` might do the same thing here since the `Default for String` constructs a new empty string - see https://doc.rust-lang.org/std/string/struct.String.html#impl-Default and https://doc.rust-lang.org/src/alloc/string.rs.html#2207-2213